shared/test/test_animations.rb

```diff
@@ -377,6 +377,23 @@ def test_doesnt_mask_unrelated_errors
     soft_delete(filename)
   end
 
+  def test_get_object_with_latest_version
+    filename = @api.randomize_filename('test.png')
+    delete_all_animation_versions(filename)
+
+    # Create an animation file
+    original_version_id = upload(filename, 'stub-v1-body')
+
+    # Delete it.
+    soft_delete(filename)
+
+    # Attempt to get object with special key to get latestVersion
+    response = AnimationBucket.new.get(@channel_id, filename, nil, 'latestVersion')
+    assert_equal response[:version_id], original_version_id
+
+    soft_delete(filename)
+  end
+
```

Comment:
Good test!  It would be nice to have at least two versions of the animation (in addition to the deleted one) to prove that we're getting the latest nondeleted version, and not the earliest nondeleted version.

```rb
v1 = upload(filename, 'v1-body')
v2 = upload(filename, 'v2-body')
soft_delete(filename)
response = AnimationBucket.new.get(@channel_id, filename, nil, 'latestVersion')
assert_equal response[:version_id], v2
```

I'd also cover the following cases:

- The simpler case: Asking for the latest version of a non-deleted animation gets you the latest version
  ```rb
  v1 = upload(filename, 'v1-body')
  v2 = upload(filename, 'v2-body')
  response = AnimationBucket.new.get(@channel_id, filename, nil, 'latestVersion')
  assert_equal response[:version_id], v2
  ```
- A failure case: Asking for the latest version of a missing/hard-deleted animation gets you a 404
  ```rb
  upload(filename, 'v1-body')
  delete_all_animation_versions(filename)
  response = AnimationBucket.new.get(@channel_id, filename, nil, 'latestVersion')
  assert_equal response[:status], 'not_found' # or whatever the equivalent is...
  ```